scripts/rag_evaluation/utils/evaluate_retrieval.py

```diff
@@ -0,0 +1,159 @@
+                            recall_sum += 1  # Since there's only one relevant document
+
+    # Calculate final metrics
+    retrieval_accuracy = total_contexts_found / total_questions if total_questions > 0 else 0
+    average_position = position_sum / total_contexts_found if total_questions > 0 else 0
+    mrr = reciprocal_rank_sum / total_questions if total_questions > 0 else 0
+    average_precision = precision_sum / total_questions if total_questions > 0 else 0
+    average_recall = recall_sum / total_questions if total_questions > 0 else 0
+
+    return {
+        "retrieval_accuracy": retrieval_accuracy,
+        "average_position": average_position,
+        "mrr": mrr,
+        "average_precision": average_precision,
+        "average_recall": average_recall,
+        "total_questions": total_questions,
+        "total_openai_json_errors": openai_json_error,
+        "total_contexts_found": total_contexts_found,
+        "position_sum": position_sum
+    }
+
+
+if __name__ == "__main__":
+    # Define input parameters
+    parser = argparse.ArgumentParser(description="Evaluate retrieval metrics")
+
+    parser.add_argument('experiment_iteration',
+                        type=int,
+                        help='The iteration number of the current experiment in a series.')
+    parser.add_argument('num_sampled_questions',
+                        type=int,
+                        help='The number of questions to sample from the set of questions generated by OpenAI.')
+    parser.add_argument('endpoint_url',
+                        type=str,
+                        default="http://0.0.0.0:8000/search",
+                        help='The URL with the endpoint to query for search results.')
+
+    args = parser.parse_args()
+
+    experiment_iteration = args.experiment_iteration
+    num_sampled_questions = args.num_sampled_questions
+    endpoint_url = args.endpoint_url
+
+    # Initialize clearml task and endpoint URL
+    task = Task.init(project_name="Fasten",
+                     task_name=f"Retrieval evaluation {experiment_iteration}")
+
+    # Load JSON data
+    with open('../data/json_sample_parsed.json', 'r') as f:
+        parsed_data = json.load(f)
+
+    entry_dict = {str(entry["node_id"]): entry for entry in parsed_data["entry"]}
+
+    openai_responses = []
+    with open('../data/batch_3haP1i8Dsfohov5umgZanshX_output.jsonl', 'r') as f:
+        for line in f:
+            openai_responses.append(json.loads(line))
+
+    # Initialize counters
+    total_contexts = len(entry_dict)
+    total_questions = 0
+    total_contexts_found = 0
+    position_sum = 0
+    k = 5
+
+    # Evaluate metrics
+    metrics = calculate_retrieval_metrics(openai_responses,
+                                          num_sampled_questions,
+                                          endpoint_url)
+
+    metrics["Total contexts"] = total_contexts
+
+    for series_name, value in metrics.items():
+        task.get_logger().report_scalar(
```

Comment:
You can also use logger.report_single_value(metric, value) and results will be shown in a table of metrics, instead of a chart by iteration, as you have no iterations here